src/seedu/addressbook/data/person/Contact.java

```diff
@@ -0,0 +1,30 @@
+package seedu.addressbook.data.person;
+
+import seedu.addressbook.data.exception.IllegalValueException;
+
+public abstract class Contact {
+    protected String trimmedValue;
+    public String value;
+    protected boolean isPrivate;
+
+    protected Contact(String value, boolean isPrivate) throws IllegalValueException {
+        this.isPrivate = isPrivate;
+        this.value = value;
+        this.trimmedValue = value.trim();
+    }
+
+    @Override
+    public String toString() {
+        return value;
+    }
+
+    @Override
+    public int hashCode() {
```

Comment:
Check out what the `hashCode` method does. Placing it in the `Contact` class would not make sense
(hint: `hashCode` is a method that needs to be overridden)